Sources/Fuzzilli/Evaluation/ProgramCoverageEvaluator.swift

```diff
@@ -121,7 +187,49 @@ public class ProgramCoverageEvaluator: ComponentBase, ProgramEvaluator {
             return true
         }
     }
-    
+
+    func resetEdge(_ edge: UInt32) {
+        resetCounts[edge] = (resetCounts[edge] ?? 0) + 1
+        if resetCounts[edge]! <= maxResetCount {
+            libcoverage.cov_clear_edge_data(&context, UInt64(edge))
+        }
+    }
+
+    public func resetAspects(_ aspects: ProgramAspects) {
+        let edgeSet = aspects as! CovEdgeSet
+        for edge in edgeSet.toEdges() {
+            resetEdge(edge)
+        }
+    }
+
+    // Evaluates the provided exection, and calculates the intersection of the coverage with the provided
+    // aspect. Resets any edges found by the execution but not the provided aspect.
+    // Returns a program aspect of the intersection, and a boolean indicating if the intersection is the same as the provided aspect
+    public func evaluateAndIntersect(_ execution: Execution, with aspects: ProgramAspects) -> (ProgramAspects, Bool)? {
+        guard let firstCov = aspects as? CovEdgeSet else { 
+            logger.fatal("Coverage Evaluator received non coverage aspects")
+        }
+
+        guard let secondCovEdgeSet = evaluate(execution) as? CovEdgeSet else { return nil }
+
+        let firstCovSet = Set(UnsafeBufferPointer(start: firstCov.edges, count: Int(firstCov.count)))
+        let secondCovSet = Set(UnsafeBufferPointer(start: secondCovEdgeSet.edges, count: Int(secondCovEdgeSet.count)))
+
+        // Reset any edges found in the second execution but not the first
+        let secondExecNewEdges = secondCovSet.subtracting(firstCovSet)
```

Comment:
Nit: how about merging these two lines into one: `for edge in secondCovSet.subtracting(firstCovSet) {`? Maybe a bit nicer, but fine either way